vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
@stevvooe 
`You get the same exact result if you have a single argument, since both use the complement as the oldValue.` I would rather be explicit by passing both old and new values.

 `{Enable|Disable}Service` is libnetwork API. Its not an implementation detail of swarmkit.

By idempotent what I meant here is one can call EnableService on a sandbox when the service is already enabled. If you are thinking about concurrent calls to EnableService & DisableService calls that means the healtcheck for some reason is flipping between healthy and unhealthy and has to be handled by the caller.